### DISCUSSION TITLE
Identifying Players in "Starting" Positions (Not BE/IR)

### DISCUSSION THREAD

**Discussion by MrDGer** [cwendt94/espn-api#364] [Q&A] [answered]
After checking out some of the other discussions, I was able to start identifying the "starters" in my league, but it's missing a few here and there.  Once I looked into it, it seems it's because it's players that have been added to the team after the draft and replacing players that have been moved to the IR.

So, for one team, it's not identifying the 2nd flex position because his roster number is higher than 17, but if I increase the loop to higher than 17, it comes back with a 'list index out of range' error.

   

    qb_rosternum = 0
    rb_rosternum = 0
    wr_rosternum = 0
    te_rosternum = 0
    flex_rosternum = 0
    dst_rosternum = 0
    k_rosternum = 0
    matchuprosternum = 0

    while matchuprosternum < 5:
        matchuprosterstr = str(matchuprosternum)
        
        while rosterspots < 17:
            
            if box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'QB':
                qb_rosternum += 1
                qbrosterstr = str(qb_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homeqb' + qbrosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homeqb' + qbrosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'RB':
                rb_rosternum += 1
                rbrosterstr = str(rb_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homerb' + rbrosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homerb' + rbrosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'WR':
                wr_rosternum += 1
                wrrosterstr = str(wr_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homewr' + wrrosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homewr' + wrrosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))    
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'TE':
                te_rosternum += 1
                terosterstr = str(te_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homete' + terosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homete' + terosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'RB/WR/TE':
                flex_rosternum += 1
                flexrosterstr = str(flex_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homeflex' + flexrosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homeflex' + flexrosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))                
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'D/ST':
                dst_rosternum += 1
                dstrosterstr = str(dst_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homedst' + dstrosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homedst' + dstrosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))
            elif box_scores[matchuprosternum].home_lineup[rosterspots].slot_position == 'K':
                k_rosternum += 1
                krosterstr = str(k_rosternum)
                with open('playerinfo/matchup' + matchuprosterstr + '_homek' + krosterstr + '_name.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].name))
                with open('playerinfo/matchup' + matchuprosterstr + '_homek' + krosterstr + '_pts.txt', 'w') as f:
                    f.write(str(box_scores[matchuprosternum].home_lineup[rosterspots].points))

            rosterspots += 1
        
        rosterspots = 0
        qb_rosternum = 0
        rb_rosternum = 0
        wr_rosternum = 0
        te_rosternum = 0
        flex_rosternum = 0
        dst_rosternum = 0
        k_rosternum = 0
        matchuprosternum += 1
`

I am fully aware that this is probably an extremely elongated and inefficient attempt at pulling this off, but I am not a programmer and am learning as I figure this out.  Any thoughts on how to get those players that go beyond the 17 spots?

**Comment by MrDGer** [comment, marked as answer]
Please disregard.  Just changed it to a for loop rather than a while and it's working as intended.  Thanks again for all this work!

**Reply by cwendt94** [reply]
Glad you were able to get it to work!
